src/pages/AgentNew.tsx

```diff
@@ -25,6 +46,28 @@ export default function AgentNew() {
     setOfficeSearchText('')
   }
 
+  const onSubmit: SubmitHandler<IAddAnAgentForm> = async ({
+    office,
+    ...params
+  }: IAddAnAgentForm) => {
+      const agentParams = {
+        agent: {
+          firstName: params.firstName,
+          lastName: params.lastName,
+          officeId: office.id
+        }
+      }
+      const newAgent = await ApiAgent.create(agentParams)
```

Comment:
My suggestion above in `agent.ts` means we can simplify here as well. Since the agentParams are not used anywhere else, we can pass the required data directly into the api function:
```suggestion
    const newAgent = await ApiAgent.create({
      ...params,
      officeId: office.id,
    })
```